app/Http/Controllers/ClientController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\ClientStoreRequest;
+use App\Models\Client;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Auth;
+
+class ClientController extends Controller
+{
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     *
+     */
+    public function index()
+    {
+        $clients = Client::withCount('projects')->with('projects')->orderBy('projects_count', 'desc')->paginate(10);
+        return view('clients.index', compact('clients'));
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        abort_if(!Auth::user()->hasRole('admin'), 401, __('Not Allowed action'));
+        return view('clients.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\ClientStoreRequest $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(ClientStoreRequest $request)
+    {
+        if (!$client=Client::create($request->validated())) {
+            return redirect()->back()->with(['error' => __('Create error')]);
+        }
+        return redirect()->to(url('/clients'))->with(['message' => $client->name . ' ' . __('Successfully created')]);
```

Comment:
@faxunil not sure what error you're having with route names, but please debug it, or give more details, it should be `return redirect()->route(some_route_name)`